allennlp/data/fields/multilabel_field.py

```diff
@@ -0,0 +1,129 @@
+from typing import Dict, Union, Set
+import logging
+
+from collections import defaultdict
+from overrides import overrides
+import torch
+from torch.autograd import Variable
+
+from allennlp.data.fields.field import Field
+from allennlp.data.vocabulary import Vocabulary
+from allennlp.common.checks import ConfigurationError
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+
+class MultiLabelField(Field[torch.Tensor]):
+    """
+    A ``MultiLabelField`` is an extension of the :class:`LabelField` that allows for multiple labels.
+    It is particularly useful in multi-label classification where more than one label can be correct.
+    As with the :class:`LabelField`, labels are either strings of text or 0-indexed integers (if you wish
+    to skip indexing by passing skip_indexing=True).
+    If the labels need indexing, we will use a :class:`Vocabulary` to convert the string labels
+    into integers.
+
+    This field will get converted into a vector of length equals the vocabulary size with
+    one hot encoding for the labels (all zeros, and ones for the labels).
+
+    Parameters
+    ----------
+    labels : ``Set[Union[str, int]]``
+    label_namespace : ``str``, optional (default="labels")
+        The namespace to use for converting label strings into integers.  We map label strings to
+        integers for you (e.g., "entailment" and "contradiction" get converted to 0, 1, ...),
+        and this namespace tells the ``Vocabulary`` object which mapping from strings to integers
+        to use (so "entailment" as a label doesn't get the same integer id as "entailment" as a
+        word).  If you have multiple different label fields in your data, you should make sure you
+        use different namespaces for each one, always using the suffix "labels" (e.g.,
+        "passage_labels" and "question_labels").
+    skip_indexing : ``bool``, optional (default=False)
+        If your labels are 0-indexed integers, you can pass in this flag, and we'll skip the indexing
+        step.  If this is ``False`` and your labels are not strings, this throws a ``ConfigurationError``.
+    """
+    # It is possible that users want to use this field with a namespace which uses OOV/PAD tokens.
+    # This warning will be repeated for every instantiation of this class (i.e for every data
+    # instance), spewing a lot of warnings so this class variable is used to only log a single
+    # warning per namespace.
+    _already_warned_namespaces: Set[str] = set()
+
+    # Dictionary from a vocab namespace to size. Need the vocab size to map labels to a tensor
+    _vocab_size: Dict[str, int] = defaultdict(int)
+
+    def __init__(self,
+                 labels: Set[Union[str, int]],
```

Comment:
I think this should be `Sequence` not `Set`, so that it type-checks to pass in a list (as you do in some places)

Reply:
done